TrackingPerformance/Condor/condorJobs_sim.py

```diff
@@ -1,156 +1,248 @@
-        output_dir = os.path.join(EosDir, part); os.makedirs(output_dir, exist_ok=True)
-        output_file = os.path.join(output_dir, output_file_name)
-        if check_output and os.path.exists(output_file):
-            root_file = ROOT.TFile(output_file, "READ")
-            events_tree = root_file.Get("events")
-            if events_tree:
-                if events_tree.GetEntries() == int(Nevt_per_job):
-                    root_file.Close()
-                    continue
-            root_file.Close()
-        else:
-            need_to_create_scripts = True
-        
-        time.sleep(1)
-        seed = str(time.time()%1000)
-        
-        arguments = (
-                #f" --compactFile /afs/cern.ch/work/g/gasadows/k4geo/FCCee/CLD/compact/{DetectorModelList_[0]}_3T/{DetectorModelList_[0]}.xml "
-                f" --compactFile ${K4GEO}/FCCee/CLD/compact/{DetectorModelList_[0]}/{DetectorModelList_[0]}.xml "    
-                "--outputFile " + output_file_name + " "
-                "--steeringFile " + "CLDConfig/CLDConfig/cld_steer.py "
-                "--random.seed " + seed + " "
-                "--enableGun "
-                "--gun.particle " + part + "- "
-                "--gun.energy " + momentum + "*GeV "
-                "--gun.distribution uniform "
-                "--gun.thetaMin " + theta + "*deg "
-                "--gun.thetaMax " + theta + "*deg "
-                "--crossingAngleBoost 0 "
-                "--numberOfEvents " + Nevt_per_job 
-            )
-        command = "ddsim " + arguments + " > /dev/null"
-
-        # Write bash script for job execution
-        bash_script = (
-            "#!/bin/bash \n"
-            f"source {setup} \n"
-            "git clone https://github.com/gaswk/CLDConfig.git \n"
-            f"{command} \n"
-            f"xrdcp {output_file_name} root://eosuser.cern.ch/{output_dir} \n"
+import importlib.util
+import sys
+from os import system  # for execution at the end
+from pathlib import Path
+
+import ROOT
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Script to run with user configuration."
+    )
+    parser.add_argument(
+        "--config",
+        type=Path,
+        required=True,
+        help="Path to the configuration file. The suffix '.py' can be omitted.",
+    )
+    return parser.parse_args()
+
+
+def load_config(config_name):
+
+    config_file = Path(config_name)
+
+    # Ensure the config file has the .py extension
+    if config_file.suffix != ".py":
+        config_file = config_file.with_suffix(".py")
+
+    # Create an absolute path
+    config_path = config_file.resolve()
+
+    # Ensure the file exists
+    if not config_path.exists():
+        raise FileNotFoundError(f"The configuration file {config_path} does not exist.")
+
+    spec = importlib.util.spec_from_file_location("config", config_path)
+    config = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(config)
+    return config
```

Comment:
this looks familiar, @tmadlener do we have this already importable from somewhere in key4hep and if not should we make that happen?

Reply:
We were discussing putting [this bit](https://github.com/iLCSoft/ILDConfig/blob/master/StandardConfig/production/py_utils.py#L9-L49) into the k4FWCore utils, but ILDReconstruction was the only use case at the time. We could move it to k4FWCore.utils

Reply:
It is not entirely the same. We could tackle this in a later refactoring

Reply:
Fine with me. I just wanted to avoid having to many copies of the same thing flying around. :)